aquadoggo/src/materializer/tasks/dependency.rs

```diff
@@ -1,9 +1,315 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
 use crate::context::Context;
-use crate::materializer::worker::TaskResult;
+use crate::materializer::worker::{Task, TaskError, TaskResult};
 use crate::materializer::TaskInput;
 
-pub async fn dependency_task(_context: Context, _input: TaskInput) -> TaskResult<TaskInput> {
-    Ok(None)
+pub async fn dependency_task(context: Context, input: TaskInput) -> TaskResult<TaskInput> {
+    let document_view_id = match input.document_view_id {
+        Some(id) => id,
+        // We only accept handling dependency tasks for specific document views.
+        None => return Err(TaskError::Failure),
```

Comment:
Nice, we could make it critical as well, as it would be rather in the bug category if someone would write such code / dispatch dependency tasks without the right info.

Reply:
I actually changed the behaviour here now to handle dependency tasks for document ids as well. Not sure if we need that behaviour, but it's simple enough to get the document view and document view id we need anyway so it didn't seem like a problem. If we know we never want to dispatch a dependency task by document id we can of course bring back this error.